server/test/com/cloud/vm/UserVmManagerImplTest.java

```diff
@@ -217,8 +221,42 @@ private void configureDoNothingForMethodsThatWeDoNotWantToTest() throws Resource
         Mockito.doNothing().when(userVmManagerImpl).validateInputsAndPermissionForUpdateVirtualMachineCommand(updateVmCommand);
         Mockito.doReturn(new ArrayList<Long>()).when(userVmManagerImpl).getSecurityGroupIdList(updateVmCommand);
         Mockito.doReturn(Mockito.mock(UserVm.class)).when(userVmManagerImpl).updateVirtualMachine(Mockito.anyLong(), Mockito.anyString(), Mockito.anyString(), Mockito.anyBoolean(),
-                Mockito.anyBoolean(), Mockito.anyLong(),
-                Mockito.anyString(), Mockito.anyBoolean(), Mockito.any(HTTPMethod.class), Mockito.anyString(), Mockito.anyString(), Mockito.anyString(), Mockito.anyListOf(Long.class),
-                Mockito.anyMap());
+                Mockito.anyBoolean(), Mockito.anyLong(), Mockito.anyString(), Mockito.anyBoolean(), Mockito.any(HTTPMethod.class), Mockito.anyString(), Mockito.anyString(),
+                Mockito.anyString(), Mockito.anyListOf(Long.class), Mockito.anyMap());
+    }
+
+    @Test
+    public void lookForNextAvailableMacAddressInNetworkIfMacIsInvalidTestMacAddressNotNull() throws InsufficientAddressCapacityException {
+        configureGetNextAvailableMacAddressInNetworkIfMacIsNull(0, "00:00:00:00:00:00", "00:00:00:00:00:00");
+    }
+
+    @Test
+    public void lookForNextAvailableMacAddressInNetworkIfMacIsInvalidTestMacAddressNull() throws InsufficientAddressCapacityException {
+        configureGetNextAvailableMacAddressInNetworkIfMacIsNull(1, null, "00:00:00:00:00:00");
+    }
+
+    @Test
+    public void lookForNextAvailableMacAddressInNetworkIfMacIsInvalidTestMacAddressBlank() throws InsufficientAddressCapacityException {
+        configureGetNextAvailableMacAddressInNetworkIfMacIsNull(1, " ", "00:00:00:00:00:00");
+    }
+
+    @Test
+    public void lookForNextAvailableMacAddressInNetworkIfMacIsInvalidTestMacAddressEmpty() throws InsufficientAddressCapacityException {
+        configureGetNextAvailableMacAddressInNetworkIfMacIsNull(1, "", "00:00:00:00:00:00");
+    }
+
+    @Test
+    public void lookForNextAvailableMacAddressInNetworkIfMacIsInvalidTestMacAddressNotValid() throws InsufficientAddressCapacityException {
+        configureGetNextAvailableMacAddressInNetworkIfMacIsNull(1, "00:00:00:00:00", "00:00:00:00:00:00");
+    }
+
+    private void configureLookForNextAvailableMacAddressInNetworkIfMacIsInvalid(int times, String macAddress, String expectedMacAddress) throws InsufficientAddressCapacityException {
```

Comment:
can you add tests for more realistic MACs like "01:23:45:67:89:ab" and invalid ones like "ab:cd:ef:gh:ij:kl"?